src/firmware/app/primitives/move_manager.c

```diff
@@ -0,0 +1,224 @@
+#include "firmware/app/primitives/move_manager.h"
+
+#include <assert.h>
+#include <math.h>
+#include <stdio.h>
+
+#include "firmware/app/control/bangbang.h"
+#include "firmware/app/control/control.h"
+#include "firmware/app/control/physbot.h"
+#include "firmware/shared/physics.h"
+#include "firmware/shared/util.h"
+#include "primitive.h"
+#include "shared/constants.h"
+#include "shared/robot_constants.h"
+
+// these are set to decouple the 3 axis from each other
+// the idea is to clamp the maximum velocity and acceleration
+// so that the axes would never have to compete for resources
+#define TIME_HORIZON 0.05f  // s
+
+// The minimum distance away from our destination that we must be if we
+// are going to rotate the bot onto its wheel axis
+// 2 * P_PI * ROBOT_RADIUS = robot circumference, which is approximately
+// how far the bot would have to turn for one full rotation, so we
+// set it a litle larger than that.
+const float APPROACH_LIMIT = 3 * P_PI * ROBOT_RADIUS;
+
+const float PI_2 = P_PI / 2.0f;
+
+typedef struct MoveManagerState
```

Comment:
I like the idea of the MoveManager to help reduce duplicated code. Just wondering if we can think of a better name than "manager" since I wouldn't initially assume it actually does most of the movement.

I understand why we can't just have "move with autochip" call the regular MovePrimitive (MovePrimitive disables autokick/chip for safetyand correctness), so just feel like a slightly different name would be helpful here.

random ideas:
* CoreMove / MoveCore
* MoveHelper

Reply:
changed to MoveHelper

Reply:
The thought is that MoveHelper will soon be replaced by trajectory planner and movepositionparams will be replaced by a trajectory planner proto

Reply:
I'm a bit confused why we don't just have a `move_primitive` that moves without autokick or autochip, then the `move_with_autokick` and `move_with_autochip` primitives can just use the `move_primitive`.

Reply:
move_primitive would take MovePrimitiveMsg so it would require move_with_autokick to convert its AutokickMovePrimitiveMsg into MovePrimitiveMsg

Reply:
Gonna run into the same issue as https://github.com/UBC-Thunderbots/Software/pull/1645#discussion_r464014504 if I turn it into a MovePrimitive and we can separate "moving" from "moving and dribbling", so not changing

Reply:
Another reason it wouldn't work @garethellis0 is that I assumed the MovePrimitive would disable autokick and autochip (because that's more correct and would prevent issues if shoot didn't disable it or something). However because the MovePrimitive would be called by MoveWithAutokick, it would disable autokick after it was set, so we'd never be able to move with autokick on.